virttest/libvirt_xml/vm_xml.py

```diff
@@ -3830,3 +3836,77 @@ def __init__(self, virsh_instance=base.virsh):
                                  tag_name='gid')
         super(VMIDMapXML, self).__init__(virsh_instance=virsh_instance)
         self.xml = '<idmap/>'
+
+
+# Sub-element of OS XML
+class VMOSFWXML(base.LibvirtXMLBase):
+    """
+    Firmware tag XML class of OS tag
+
+    Elements:
+        feature: list of dict - enabled, name
+    Example:
+        <firmware>
+        <feature enabled='yes' name='enrolled-keys'/>
+        <feature enabled='yes' name='secure-boot'/>
+        </firmware>
+    """
+
+    __slots__ = ('feature',)
+
+    def __init__(self, virsh_instance=base.virsh):
+        """
+        Create new VMOSFWXML instance
+        """
+        accessors.XMLElementList("feature", self, parent_xpath='/',
+                                 marshal_from=self.marshal_from_feature,
+                                 marshal_to=self.marshal_to_feature)
+        super(VMOSFWXML, self).__init__(virsh_instance=virsh_instance)
+        self.xml = '<firmware/>'
+
+    # Sub-element of firmware
+    class FeatureXML(VMXML):
```

Comment:
Can we move FeatureXML definition outside VMOSFWXML class scope, rather than being inner class ? this may help FeatureXML reusable by others.

Reply:
I think FeatureXML may not be used by others, because feature element must be under the firmware element. Like:
<firmware>
    <feature enabled='yes' name='enrolled-keys'/>
    <feature enabled='yes' name='secure-boot'/>
</firmware>

Reply:
```suggestion
    class FeatureXML(base.LibvirtXMLBase):
```

Reply:
> Can we move FeatureXML definition outside VMOSFWXML class scope, rather than being inner class ? this may help FeatureXML reusable by others.

I disagree. I don't think we're suppose to **"reuse"** FeatureXML anywhere else, it's not like address.